File: CH1-input-process-output/EX10-stock-transaction/source-code/stockTransaction.js
// let dollarUS = Intl.NumberFormat("en-US");

// Input
var commission = 0.02,
  stock = 0,
  purchasedStock = 1000,
  purchasePrice = 32.87,
  boughtCommission,
  totalPurchase,
  soldStock = 1000,
  sellPrice = 33.92,
  soldCommission,
  totalSell,
  total;

// Process
stock = purchasedStock;
totalPurchase = purchasePrice * stock;
boughtCommission = totalPurchase * commission;

totalSell = sellPrice * stock;
soldCommission = totalSell * commission;
stock = purchasedStock - soldStock;

total = totalSell - totalPurchase - boughtCommission - soldCommission;

// Output
console.log("Total Purchased             : $" + totalPurchase.toFixed(2));
console.log("Broker commission(purchase) : $" + boughtCommission.toFixed(2));
console.log("Sold stock                  : " + soldStock);
console.log("Broker commission(sold)     : $" + soldCommission.toFixed(2));
console.log("Total profit or loss        : $" + total.toFixed(2));

File: CH1-input-process-output/EX10-stock-transaction/source-code/stockTransaction.java
public class stockTransaction {
    public static void main(String[] args){
        // Input        
        double commission = 0.02;
        int stock = 0;

        int purchasedStock = 1000;
        double purchasePrice = 32.87;
        int boughtCommission;
        int totalPurchase;

        int soldStock = 1000;
        double sellPrice = 33.92;
        int soldCommission;
        int totalSell;
        
        int total;

        // Process
        double temp;
        stock = purchasedStock;
        temp = purchasePrice * stock;
        totalPurchase = (int) temp;
        temp = totalPurchase * commission;
        boughtCommission = (int) temp;

        temp = sellPrice * stock;
        totalSell = (int) temp;
        temp = totalSell * commission;
        soldCommission = (int) temp;
        stock = purchasedStock - soldStock;

        temp = totalSell - totalPurchase - boughtCommission - soldCommission;
        total = (int) temp;

        //Output
        System.out.println("Total purchased             : $" + totalPurchase);
        System.out.println("Broker commission(purchase) : $" + boughtCommission);
        System.out.println("Sold stock                  :  " + soldStock);
        System.out.println("Broker commission(sold)     : $" + soldCommission);
        System.out.println("Total profit or loss        : $" + total);

    }
}